map/framework.cpp

```diff
@@ -3313,3 +3316,79 @@ void Framework::OnPowerSchemeChanged(power_management::Scheme const actualScheme
   if (actualScheme == power_management::Scheme::EconomyMaximum && GetTrafficManager().IsEnabled())
     GetTrafficManager().SetEnabled(false);
 }
+
+bool Framework::ShouldShowProducts() const
+{
+  auto const connectionStatus = Platform::ConnectionStatus();
+  if (connectionStatus == Platform::EConnectionType::CONNECTION_NONE)
+    return false;
+
+  std::string donateUrl;
+  if (!settings::Get(settings::kDonateUrl, donateUrl)) // donation is disabled
+    return false;
+
+  if (!storage::IsPointCoveredByDownloadedMaps(GetCurrentPlacePageInfo().GetMercator(), m_storage, * m_infoGetter))
+    return false;
+
+  #ifdef DEBUG
+  uint32_t constexpr kMinTotalForegroundTimeout = 30;
+  uint32_t constexpr kMinSessionsCount = 3;
+  uint32_t constexpr kPopupCloseTimeout = 10;
+  uint32_t constexpr kProductSelectTimeout = 20;
+  #else
+  uint32_t constexpr kMinTotalForegroundTimeout = 60 * 15; // 15 min
+  uint32_t constexpr kMinSessionsCount = 5;
+  uint32_t constexpr kPopupCloseTimeout = 60 * 60 * 24  * 14; // 14 days
+  uint32_t constexpr kProductSelectTimeout = 180 * 24 * 60 * 60; // 180 days
+  #endif
+
+  auto const now = base::SecondsSinceEpoch();
+
+  if (m_usageStats.GetTotalForegroundTime() < kMinTotalForegroundTimeout ||
+      m_usageStats.GetSessionsCount() < kMinSessionsCount)
+    return false;
```

Comment:
Let's take out the function:
```
bool UsageStats::IsLoyalUser() const
{
#ifdef DEBUG
  uint32_t constexpr kTotalForegroundTimeout = 30;
  uint32_t constexpr kSessionsCount = 3;
#else
  uint32_t constexpr kTotalForegroundTimeout = 60 * 20; // 20 min
  uint32_t constexpr kSessionsCount = 10;
#endif

  return m_sessionsCount >= kMinSessionsCount && m_totalForegroundTime >= kTotalForegroundTimeout;
}
```

And call it before IsPointCoveredByDownloadedMaps

Reply:
Nice idea! Thanks!